core/modules/simpletest/backdrop_web_test_case_cache.php

```diff
@@ -42,7 +42,7 @@ public function isCached(){
   public function prepareCache() {
     $this->setUp();
     // Speed up test speed by converting tables to MyISAM.
-    $this->alterToMyISAM();
+    //$this->alterToMyISAM();
```

Comment:
This is probably not necessary, and it will cause the test runs to be 3-4x slower.

Reply:
This may be true, but it seems that it may cause issues when MySQL8+ is used. See https://www.percona.com/blog/2016/10/11/mysql-8-0-end-myisam (article written in Oct. 2016)

> In MySQL 8.0 (DMR version as of writing), the _MyISAM storage engine is still available_. But in a very _limited_ scope:
>
> - After introducing the new data dictionary, the MyISAM tables are gone from the system schema (“mysql” db).
> - Working with MyISAM is harder now (and discouraged): you can’t just copy MyISAM tables into a running MySQL server, they will not be discovered (unlike InnoDB, where you can use “ALTER TABLE … IMPORT TABLESPACE”)
> - However, you can create a table engine=MyISAM, and it will work as before.

...

> So the only MyISAM advantages left are:
>
> - Tables will be smaller on disk compared to uncompressed InnoDB tables.
> - The count(*) is still much faster in MyISAM

...
>I would not use MyISAM unless there is a specific case, and for well-known reasons (MyISAM are non-transactional, table-level locks, with no crash recovery, etc.)

The [MySQl 8 reference manual](https://dev.mysql.com/doc/refman/8.0/en/myisam-storage-engine.html) mentions:

> MyISAM is based on the older (**and no longer available**) ISAM storage engine but has many useful extensions.

Reply:
That is true, but the test bot in ZenCI uses an older version of MySQL, so we can still use MyISAM, which is MUCH faster when dropping and creating tables.
Perhaps it would be a good idea to check which version of MySQL is being used and then change the behavior based on that.

Reply:
...also: https://www.liquidweb.com/kb/mysql-performance-myisam-vs-innodb

> ## Should I use InnoDB with WordPress, Magento or Joomla Sites?
> **The short answer here is yes, in most cases**. ... Most users of popular third-party applications like WordPress, Magento, and Joomla have limited knowledge of the underlying database components or code involved to make an informed decision on storage engines. Most table-locking bottlenecks from these content management systems (CMS) are generally resolved by changing all the tables for the site over to  InnoDB instead of the default MyISAM.  If you are hosting many of these types of CMS on your server, it would be beneficial to change the default storage engine in MySQL to use InnoDB for all new tables so that any new table installations start off with InnoDB.

Reply:
I didn't mean to include this change in this PR, it's just that I was running the tests locally and found I could not run them on MySQL 8 while that line was in place. IMO we should split that out into a different issue.